docs/design/w3c/w3c-representation.md

```diff
@@ -0,0 +1,387 @@
+## Design for W3C representation of AnonCreds credentials and presentation
+
+Currently `anoncreds-rs` library only provides support for Indy styled AnonCreds credentials matching to the [specification](https://hyperledger.github.io/anoncreds-spec/). 
+
+This design document proposes extending of `anoncreds-rs` library to add support for AnonCreds W3C representation of verifiable credential and presentation described in the [document]().
+
+### Goals and ideas
+
+* Use Indy styled credentials to generate W3C AnonCreds credentials and presentations
+  * Credentials conversion:
+    * Convert Indy styled AnonCreds credentials into W3C form
+    * Convert W3C styled AnonCreds credentials into Indy form
+  * Presentation conversion (Optional):
+    * Convert Indy styled AnonCreds presentation into W3C form
+    * Convert W3C styled AnonCreds presentation into Indy form
+* Extend W3C credentials: 
+  * Ability to set Data Integrity proof signatures for generated W3C credential objects: 
+    * W3C credentials may contain multiple signatures 
+    * AnonCreds-Rs only generates/handle AnonCreds signatures  
+
+#### Out of scope
+
+* Credentials: Verify validity of the Data Integrity proof signature
+* Presentations: Create presentation using Data Integrity proof signature
+* Presentations: Verify validity of presentations using Data Integrity proof signatures 
+* Presentations: Support different formats (for example DIF) of Presentation Request
+
+### Question impacting the approach
+
+* Q1: Do we need conversion for intermediate messages to make them W3C compliant: Credential Offer, Credential Request?
+  * Q1.1: (Depends on answer for Q1) If no conversion: Do we want Credential Offer indicates what form of credential will be issued as the process execution result?
+* Q2: Do we want duplicate methods (like `sign` and `sign_w3c`) or only use single conversion method (like `credential.to_w3c`) doing extra step?
+* Q3: Should we care about the legacy credential migration and compatibility? Or achieve better following to the W3C standard with preserving power of CL signatures?
+  * example pros of dropping legacy support:
+    * different attributes types can be used in credentialSubject (support nested objects, array, other features)
```

Comment:
This would be a nice feature, but I think we will not be able to create a mapping anymore between anoncreds and w3c. Unless we encode the nested objects, but I am not the biggest fan of that.

Reply:
I'd also assume we can not do selective disclosures / predicates over those nested attributes, or can we?